feat: show contract call arguments in a more approchable way

lukeromanowicz opened the pull request:
<!--- Ensure the title of the Pull Request follows conventional commits syntax. If it resolves an issue, provide its ID in the scope section. -->

## Description
Resolves #142 

MAINNET:
Example of AEX-9 contract with non-decoded arguments: ct_2Hyt9ZxzXra5NAzhePkRsDPDWppoatVD7CtHnUoHVbuehwR8Nb
Example of contract with decoded arguments: ct_2U1usf3A8ZNUcZLkZe5rEoBTxk7eJvk9fcbRDNqmRiwXCHAYN

TESTNET:
Example of minting NFTs: ct_cVbugNVmdz2e99bu7fiBHsJY1NhRHh53CQBjNTdaxh8pSKHgK
Example of minting NFTs using templates: ct_Wr8AH1LDaya9niRfH92atd8LerNUYVcFdYrvcTdriMrqxNzvs

## Demo
https://github.com/aeternity/aescan/assets/46789227/1ce5b57a-546c-484c-b906-ff36ddee5e6f


AEX-141 tokens (these tokens by design don't have symbols - only id, that's why it's not present in the chips):
![image](https://github.com/aeternity/aescan/assets/46789227/95776472-2c57-46b8-a1cb-137e80508697)
![image](https://github.com/aeternity/aescan/assets/46789227/95a659e0-3e93-4a32-883d-c3df8c4cceb0)



## Checklist:
<!--- Put an `x` in all the boxes that apply. -->
<!--- If your change requires a documentation PR, please link it appropriately -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] I have read and followed the [Contributing Guide](../../CONTRIBUTING.md)  
- [x] My change does not require a change to the documentation.

lukeromanowicz commented on src/app.vue:
```diff
@@ -97,6 +97,10 @@ if (import.meta.env.MODE !== 'production') {
   })
 }
+
+function logError(error) {
```
unrelated improvement to show error details in the console

michele-franchi reviewed (commented):
I had a quick look at the demo and I noticed known AEX-N events are not displayed. If you check this comment https://github.com/aeternity/aescan/issues/142#issuecomment-1693056178 one of the requirements is to also display known AEX-N events in the usual way with the mix of expandable buttons for events that cannot be decoded.

janmichek commented on src/components/ContractEventDataPanel.vue:
```diff
@@ -0,0 +1,56 @@
+    </div>
+    <dl>
+      <template
+        v-for="(arg, index) in event.data"
```
```suggestion
        v-for="(arguments, index) in event.data"
```

I would personally better read arguments than arg, if there are no technical limitations for shortening

lukeromanowicz replied:
Right, fixed :+1:

janmichek reviewed (changes requested):
Code looks good and works nice as well 👍 

When I look at the design, there are few misalignemnts

![image](https://github.com/aeternity/aescan/assets/15363559/d46573c4-25d8-4e10-a692-16ad92d27bfd)

1. the copy button is not implemented. Is this intentional? Please add it or negotiate the change in figma
2. the styles of expanded rows differs (row spacingand border radius). I better like our version. Can you negotiate the change?
3. The button label differs. Can you negotiate the change?

lukeromanowicz commented:
@michele-franchi thank you for your review. I implemented the missing features of showing decoded values for known events.

@janmichek Very good eye! I added the missing copy feature and changed the existing code to reflect the recent design changes. Unfortunately some of the changes are inconsistent with the rest of the application so I suggested some improvements in the designs. So far no response but I have reasonable confidence that they will be eventually approved. We can wait until they are approved or open another task if they won't.

michele-franchi commented on src/components/ContractEventsTable.vue:
```diff
@@ -26,62 +26,107 @@
+          <datetime-label :datetime="event.created"/>
+        </td>
+        <td :class="[{'contract-events-table__data--expanded': isOpened.includes(index)}]">
+          {{ event.eventName ? event.eventName : 'N/A' }}
```
```suggestion
          {{ event.eventName || 'N/A' }}
```

michele-franchi reviewed (approved):
I found some minor issues but it looks good, I approve in advance.

CharisSiarampalis commented:
@lukeromanowicz When i try to click on some smart contracts i can see the page loading up but then nothing appears
Example:
https://pr-486-aescan.stg.aepps.com/contracts/ct_rtqLKcZfQSx1GjUVBkVtRjmTGf3E2DLrSYYcaQsNYNrEtTyN1
![image](https://github.com/aeternity/aescan/assets/42606212/2042e44e-f25e-42fe-9374-c41e433adcd0)

On dev  i can see it
https://aescan.dev.aepps.com/contracts/ct_rtqLKcZfQSx1GjUVBkVtRjmTGf3E2DLrSYYcaQsNYNrEtTyN1

lukeromanowicz commented:
@CharisSiarampalis that's an interesting case. It's not an AEX-x compliant token and yet it's decoded. Weird. Nonetheless fixed. Thanks for reporting

CharisSiarampalis commented:
@lukeromanowicz In some cases i see N/A for name.
Is this normal?
![image](https://github.com/aeternity/aescan/assets/42606212/ad737d0a-d4b8-4499-ac4d-5c68e971c7ac)

lukeromanowicz commented:
@CharisSiarampalis correct, not all of the events can be decoded by the middleware

janmichek commented on src/components/ExpandButton.vue:
```diff
@@ -1,9 +1,9 @@
 <template>
   <div>
```
is the `div` wrap needed?

lukeromanowicz replied:
Absolutely not. Thanks :+1:

janmichek reviewed (changes requested):
Nice work! User-wise I am satisfied. Giving some code suggestions. Let me know what you think.

janmichek commented:
The copy icons does not seem to be aligned to anything
![image](https://github.com/aeternity/aescan/assets/15363559/fabc3e43-0a8e-490a-a1a2-1521137b6566)

janmichek commented:
![image](https://github.com/aeternity/aescan/assets/15363559/9bae4da2-ee0a-4b91-819f-947e0d934744)

Probably not related to the feature, but could you add some whitespace  between the last row and pagination? Seems to be inconsistent with other tables

janmichek commented:
Could you utilize better the space for arguments value here?

![image](https://github.com/aeternity/aescan/assets/15363559/097d5cc4-4d41-43e9-ac8b-7a20101c5851)

lukeromanowicz commented on src/components/ContractEventCell.vue:
```diff
@@ -0,0 +1,54 @@
+      return await import(`@/components/ContractEventCell${props.event.eventName}.vue`)
+    } catch {
+      console.error(`Uknown event ${props.event.eventName} in contract ${props.contractDetails.id}`)
+      return defineComponent(() => () => h('span', 'N/A'))
```
Since computed property should not modify another variable when getting read, introducing hasError here would be against good practices of vue clean code. We would either need to catch this error in the parent components (where ContractEventCell is used) or define a new vue component that in the template simply shows <span>N/A</span>. I considered all of these options and thought that the currently used approach makes the most sense and is simplistic.

janmichek replied:
Thanks for an explanation, I think it makes sense. Could you point me to that good practices of vue clean code?

lukeromanowicz commented on src/components/ContractEventCellTemplateCreation.vue:
```diff
@@ -0,0 +1,18 @@
+
+<script setup>
+const props = defineProps({
+  contractDetails: {
```
Some of the Event Cells need it. If I remove this in the cells when this is unused, vue will start sending warnings that a value was passed to a property that doesn't exist :/

janmichek replied:
I see, but why is this labeled as `required` then?

lukeromanowicz commented:
Very good eye and great suggestions @janmichek! Thank you! I've applied all of the recommendations except those on which I commented. Please let me know what you think

janmichek reviewed (changes requested):
I found one more issue. See this scenario

https://github.com/aeternity/aescan/assets/15363559/4ac46959-d2f0-451e-8578-59d4a92eeb9a

lukeromanowicz commented:
@janmichek it's a middleware issue throwing 500. We are (not) handling it exactly the same in every other table of the app. How do you suggest to fix it?

janmichek commented:
> @janmichek it's a middleware endpoint issue which is throwing 500. We are (not) handling it exactly the same in every other table of the app. How do you suggest to fix it?

If it's a middleware issue, then I would report it and create some followup on our side

lukeromanowicz commented:
Reported to the MDW and created a followup on our side: https://github.com/aeternity/aescan/issues/498
